src/main/java/com/github/jreddit/parser/listing/SubmissionsListingParser.java

```diff
@@ -0,0 +1,56 @@
+package com.github.jreddit.parser.listing;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jreddit.parser.entity.Submission;
+import com.github.jreddit.parser.entity.Thing;
+import com.github.jreddit.request.error.RedditError;
+
+public class SubmissionsListingParser extends RedditListingParser {
```

Comment:
I've tried this, but there are two problems:

* Generic types are lost after compile-time, thus casting to generic type depended on the kind that is provided cannot working.
* There two are listings which have two types of things: in a comment tree Comment/More, and in a user overview Comment/Submission

I don't know how to get around this without a little code duplication.